src/index.ts

```diff
@@ -1,17 +1,17 @@
 #!/usr/bin/env node
 import { Cli } from './cli';
 import { DependencyResolver } from './dependency-resolver';
+import { DryCommandInterceptor } from './dry-command-interceptor';
 import { DryPackage } from './dry-package';
-import { NpmCommandProxy } from './npm-command-proxy';
 
 const cli = Cli.of(process);
-const npmCommandProxy = new NpmCommandProxy(cli, process);
+const dryCommandInterceptor = new DryCommandInterceptor(cli, process);
 const dependencyResolver = new DependencyResolver(cli);
 
 DryPackage.readFromDisk(dependencyResolver)
     .buildNpmPackage()
     .then((npmPackage) => {
         npmPackage.beforeNpmRun();
-        return npmCommandProxy.proxy().then(() => npmPackage.afterNpmRun());
+        return dryCommandInterceptor.proxy().then(() => npmPackage.afterNpmRun());
```

Comment:
In this case, `npmPackage.beforeNpmRun` and `npmPackage.afterNpmRun` should be runned by the command interceptor itself.

Reply:
Ok, done. The interceptor is now an executor responsible for executing the dry command.
It also manages specific dry configuration parameters.
NpmPackage now implements functions like save/update/delete instead of beforeNpmRun/afterNpmRun.